src/cli.js

```diff
@@ -201,7 +211,14 @@ Promise.resolve().then(() => {
   if (!output) { return }
   process.stdout.write(output)
   if (errored) { process.exitCode = 2 }
-}).catch(err => {
-  console.log(err.stack) // eslint-disable-line no-console
-  process.exit(err.code || 1)
+}).catch(({ code, message, stack }) => {
+  const exitCode = code === undefined ? 1 : code
+
+  if (code > 0) {
+    process.stdout.write(`${stack}${EOL}`)
```

Comment:
@finico squash your commits in one for best history and convenient review in future :+1: